src/acvp_dsa.c

```diff
@@ -338,18 +345,37 @@ static ACVP_RESULT acvp_dsa_output_tc (ACVP_CTX *ctx, ACVP_DSA_TC *stc, JSON_Obj
         switch (stc->gen_pq) {
         case ACVP_DSA_CANONICAL:
         case ACVP_DSA_UNVERIFIABLE:
-            json_object_set_string(r_tobj, "g", (char *) stc->g);
+            tmp = calloc(ACVP_DSA_PQG_MAX, sizeof(char));
+            rv = acvp_bin_to_hexstr(stc->g, stc->g_len, (unsigned char *) tmp);
```

Comment:
Check stc->g_len before doing the conversion. Use could have set it to some weirdly large value. Example code in acvp_kdf108.c